.github/workflows/gems-bump-version.yml

```diff
@@ -15,7 +15,17 @@ jobs:
   Create-PR-To-Bump-Dependabot-Gems-Version:
     runs-on: ubuntu-latest
     steps:
+      - name: Generate token
+        id: generate_token
+        uses: tibdex/github-app-token@b62528385c34dbc9f38e5f4225ac829252d1ea92 # v1.8.0
+        with:
+          app_id: ${{ secrets.DEVELOPER_AUTOMATION_APP_ID }}
+          private_key: ${{ secrets.DEVELOPER_AUTOMATION_PRIVATE_KEY }}
+
       - uses: actions/checkout@v3
+        with:
+          token: ${{ steps.generate_token.outputs.token }}
+
```

Comment:
I believe that checkout step can continue to use the generated token

Reply:
It can, but a few steps later we have:

1. Create a branch, commit the changes
2. Push the branch

I'm not actually sure if 1️⃣ requires a write token, since it's a local branch/commit, but I am sure that 2️⃣ needs the token... 

And I couldn't find a way to pass the token to `git` in a non-clunky way... rather than adding something like https://stackoverflow.com/a/57932145/770425, I thought it was simpler to just `checkout` the repo with the write permissions and then I didn't need to doublecheck if 1️⃣ needed the token and I didn't need to configure custom tokens in the URLs... 

Technically it might be slightly less secure, but there's only two steps that get elevated privileges beyond what they need, and that's bumping the version and configuring the git user... so I thought it was fine to take the shortcut in the name of visual simplicity. 😀

Reply:
What I meant is that the checkout step specifically does not seem to need the elevated token and you're passing the `GH_TOKEN` to the GH cli manually. 

You can still generate it and use it for the other step unless I am missing something. It does not change anything in terms of security, but I was just curious

Reply:
I see what you mean, I missed the steps where you call `git` directly

Thank you for explaining!